staging/src/k8s.io/apiserver/pkg/authentication/request/websocket/protocol.go

```diff
@@ -106,3 +106,7 @@ func (a *ProtocolAuthenticator) AuthenticateRequest(req *http.Request) (*authent
 
 	return resp, ok, err
 }
+
+func protocolAuthError(err error) error {
+	return authenticator.NewError("websocket", err)
```

Comment:
ProtocolAuthenticator delegates to `auth authenticator.Token` for authentication... "websocket" as an authenticator name isn't very meaningful

Reply:
That is correct.
However, before ProtocolAuthenticator calls into authenticator.Token, it validates several preconditions which may result in errors. 
Should we attribute such errors to authenticator.Token instead?

Reply:
I guess if this is only used for error conditions specific to the websocket protocol method, that's ok